src/Simplex/Messaging/Agent/Transmission.hs

```diff
@@ -103,60 +107,82 @@ data SMPMessage
         previousMsgHash :: ByteString,
         agentMessage :: AMessage
       }
+  deriving (Show)
 
 data AMessage where
   HELLO :: VerificationKey -> AckMode -> AMessage
   REPLY :: SMPQueueInfo -> AMessage
   A_MSG :: MsgBody -> AMessage
+  deriving (Show)
 
 parseSMPMessage :: ByteString -> Either ErrorType SMPMessage
-parseSMPMessage _ = Left INTERNAL
+parseSMPMessage =
+  first (const $ SYNTAX errBadMessage)
+    . A.parseOnly (smpMessageP <* A.endOfLine <* A.endOfInput)
```

Comment:
reuse `parse` from #184 here?

Reply:
good spot - I missed - will change in the next PR.